sycl/doc/design/CompileTimeProperties.md

```diff
@@ -769,9 +872,10 @@ define spir_kernel void @MyKernel(%arg1, %arg2) !spirv.ParameterDecorations !0 {
 
 When a property on a global variable needs to be represented in SPIR-V, we
 generally translate the property into a SPIR-V **OpDecorate** instruction for
-the corresponding module scope (global) **OpVariable**.  Again, there is no
-existing mechanism to do this in the SPIR-V LLVM Translator, so we propose the
-following mechanism.
+the corresponding module scope (global) **OpVariable**. If the property is 
+applied to any of the nested member variables of the global variable it will
+still be translated into a SPIR-V **OpDecorate** instruction but will just be
+applied to the result of a SPIR-V **SpecConstantOp** call.
```

Comment:
I think this section needs elaboration.  What SpecConstantOp call is this being applied to?  More generally, given the format of llvm.global.annotations, it seems we can't tell the difference between an annotation on the global memory itself and one on its first member.  This seems to be because this language feature was never updated for opaque pointers (probably because it was forgotten about since it wasn't in the LangRef).  Thus it seems that we need to first fix LLVM's global annotations for opaque pointers before making these improvements.